src/odemis/acq/stitching/_registrar.py

```diff
@@ -726,28 +726,21 @@ def _assemble_mosaic(self):
                     adj.append([j, shifts[j][i]])
             adj_list[i] = adj
 
-        positions = {0: (0, 0)}
+        positions = numpy.zeros((num_rows, num_cols, 2))  # start with no shift for each tile
+        positions_flat = positions.reshape((num_rows * num_cols, 2))  # the tiles in the "idx" notation
         idx_queue = deque([0])  # start with index 0
+        upd_positions = []  # list of positions that have already been updated
```

Comment:
I'd suggest to name it something like `idx_done`, to make it a little bit more symetrical with `idx_queue`. Also, make it a set, as it's a little faster, and conveys the fact it the order doesn't matter.

Reply:
Good idea! I've updated the commit.